content/webapp/components/RelatedWorks.tsx

```diff
@@ -117,41 +156,81 @@ function getRelatedTabConfig({
         'production.dates.from': dateRange.from,
         'production.dates.to': dateRange.to,
       },
+      aggregations: [],
       related: relatedWorks['date-range'],
       setRelated: (results: WorkBasic[]) =>
         setRelatedWorks(prev => ({ ...prev, 'date-range': results })),
     };
   }
 
+  // If we have genres labels, we can to include a genre label (type/technique) tab.
+  // We pick the first genre label that we consider to be visual if there is one,
+  // otherwise we pick the first genre label.
+  if (genresLabels.length > 0) {
+    const visualGenre = genresLabels.find(label =>
+      visualGenres.includes(label)
+    );
+    const chosenGenre = visualGenre || genresLabels[0];
+    const id = toHtmlId(chosenGenre);
+    config[`genres-${id}`] = {
+      text: chosenGenre,
+      params: {
+        'genres.label': [chosenGenre],
+        // The genres we chose comes from what is available when using the first subject labels as a filter
+        // so we also wanter to filter here by the subject labels as well as the genre label
+        'subjects.label': subjectLabels.map(label => `"${label}"`),
+      },
+      aggregations: [],
+      related: relatedWorks[`genres-${id}`],
+      setRelated: (results: WorkBasic[]) =>
+        setRelatedWorks(prev => ({ ...prev, [`genres-${id}`]: results })),
+    };
+  }
   return config;
 }
 
 const RelatedWorks: FunctionComponent<Props> = ({ work }) => {
   const [relatedWorks, setRelatedWorks] = useState<{
     [key: string]: WorkBasic[] | undefined;
   }>({});
+  const [relatedTabConfig, setRelatedTabConfig] = useState<{
+    [key: string]: {
+      text: string;
+      params: { [key: string]: string | string[] };
+      aggregations: string[];
+      related: WorkBasic[] | undefined;
+      setRelated: (results: WorkBasic[]) => void;
+    };
+  }>({});
 
-  const relatedTabConfig = getRelatedTabConfig({
-    work,
-    relatedWorks,
-    setRelatedWorks,
-  });
-
-  // Set initial tab to first subject or date-range if no subjects
+  // Set selected tab to first available tab
   const tabKeys = Object.keys(relatedTabConfig);
   const [selectedWorksTab, setSelectedWorksTab] = useState(tabKeys[0] || '');
 
   const serverData = useContext(ServerDataContext);
 
   useEffect(() => {
-    // Only fetch if we haven't already fetched results for the current tab
-    // or if the results for the current tab now include the current work
-    const related =
-      relatedTabConfig[selectedWorksTab] &&
-      relatedTabConfig[selectedWorksTab].related;
+    // Reset related works and tab config when work changes
+    setRelatedWorks({});
+    (async () => {
+      const config = await getRelatedTabConfig({
+        work,
+        relatedWorks,
+        setRelatedWorks,
+        serverData,
+      });
+      setRelatedTabConfig(config);
+    })();
+  }, [work.id]);
```

Comment:
I think something here doesn't quite work, if I click on a related work, it updates the t
abs but no content loads. If I navigate back to the previous work, it's the same. It needs a fresh reload to load content.
<img width="1334" alt="Screenshot 2025-06-02 at 14 19 13" src="https://github.com/user-attachments/assets/17eeae1d-58d1-4eff-a7a9-7269f7a7d488" />
<img width="1257" alt="Screenshot 2025-06-02 at 14 19 07" src="https://github.com/user-attachments/assets/87ed41d7-68bb-4f33-b3c4-cb6e8ec555b5" />

Reply:
Yep, seems fixed!